public/app/features/query/components/QueryEditorRowHeader.tsx

```diff
@@ -114,7 +130,7 @@ export const QueryEditorRowHeader = <TQuery extends DataQuery>(props: Props<TQue
             {validationError && <FieldValidationMessage horizontal>{validationError}</FieldValidationMessage>}
           </>
         )}
-        {renderDataSource(props, styles)}
+        {renderDataSource(props, styles, panel?.repeat)}
```

Comment:
You can repeat by row as well, so checking panel.repeat is not enough (another argument that validating like this is not great). 

better to allow all multi valued data source variables and warn if it evaluates to multiple values (given panel.scopedVars)